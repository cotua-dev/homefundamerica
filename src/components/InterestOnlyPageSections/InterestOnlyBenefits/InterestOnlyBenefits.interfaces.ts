export interface FetchInterestOnlyBenefitsImageQuery {
    interestOnlyBenefitsImage: {
        publicURL: string;
    };
}
