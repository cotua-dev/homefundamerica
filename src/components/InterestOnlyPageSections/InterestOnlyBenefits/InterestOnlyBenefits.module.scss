section#interestOnlyBenefitsSection {
    height: 60vh;

    @media screen and (max-width: 1000px) {
        height: 100%;
    }
}
