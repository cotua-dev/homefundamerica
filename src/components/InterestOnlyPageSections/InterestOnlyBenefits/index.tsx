import React, {
    FunctionComponent,
    ReactElement,
} from 'react';
import {
    useStaticQuery,
    graphql,
} from 'gatsby';
import { FetchInterestOnlyBenefitsImageQuery } from './InterestOnlyBenefits.interfaces';
import { interestOnlyBenefitsSection } from './InterestOnlyBenefits.module.scss';
import * as sharedStyles from '../../sharedStyles/shared.module.scss';

export const InterestOnlyBenefits: FunctionComponent = (): ReactElement => {
    const queryResult = useStaticQuery<FetchInterestOnlyBenefitsImageQuery>(graphql`
        query {
            interestOnlyBenefitsImage: file(name: {eq: "AdobeStock_315056612"}, extension: {eq: "jpg"}) {
                publicURL
            }
        }
    `);

    return (
        <section
            id={interestOnlyBenefitsSection}
            className={sharedStyles.loanNegative}
        >
            <div className={sharedStyles.imageWrapper}>
                <img
                    src={queryResult?.interestOnlyBenefitsImage.publicURL}
                    alt="Interest only benefits image"
                />
            </div>
            <div className={sharedStyles.contentWrapper}>
                <h2>{`Interest-Only Mortgage Benefits`}</h2>
                <p>
                    {`
                        Control the terms of your mortgage. With an interest-only
                        mortgage, you can choose to pay the interest only or make
                        a principle payment amount. This customization is helpful
                        for borrowers with fluctuating income who want more control
                        of their terms. Benefits of an Interest-Only Mortgage:
                    `}
                </p>
                <ul>
                    <li>{`Offers initial interest-only payment options`}</li>
                    <li>{`Allows for flexible monthly payments during the initial interest-only term`}</li>
                    <li>{`Gives borrowers the opportunity to make principal payments when they choose`}</li>
                    <li>{`Custom loan tailored for specific needs`}</li>
                </ul>
            </div>
        </section>
    );
}
