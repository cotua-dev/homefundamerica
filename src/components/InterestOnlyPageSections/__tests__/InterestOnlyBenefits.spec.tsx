import React from 'react';
import {
    render,
    RenderResult,
} from '@testing-library/react';
import { InterestOnlyBenefits } from '../InterestOnlyBenefits';

describe('InterestOnlyBenefits Component', (): void => {
    test('renders', (): void => {
        const { container }: RenderResult = render(<InterestOnlyBenefits/>);
        expect(container.firstChild).toMatchSnapshot();
    });
});
