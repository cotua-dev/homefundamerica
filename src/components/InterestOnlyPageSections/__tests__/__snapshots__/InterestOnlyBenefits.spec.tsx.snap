// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterestOnlyBenefits Component renders 1`] = `
<section
  id="interestOnlyBenefitsSection"
>
  <div>
    <img
      alt="Interest only benefits image"
    />
  </div>
  <div>
    <h2>
      Interest-Only Mortgage Benefits
    </h2>
    <p>
      
                        Control the terms of your mortgage. With an interest-only
                        mortgage, you can choose to pay the interest only or make
                        a principle payment amount. This customization is helpful
                        for borrowers with fluctuating income who want more control
                        of their terms. Benefits of an Interest-Only Mortgage:
                    
    </p>
    <ul>
      <li>
        Offers initial interest-only payment options
      </li>
      <li>
        Allows for flexible monthly payments during the initial interest-only term
      </li>
      <li>
        Gives borrowers the opportunity to make principal payments when they choose
      </li>
      <li>
        Custom loan tailored for specific needs
      </li>
    </ul>
  </div>
</section>
`;
